script.js

```diff
@@ -12,60 +12,169 @@ const confirmationButton = document.getElementById("confirmation-button");
 const cartItemsContainer = document.getElementById("cart-items-container");
 const cartItemsNumber = document.getElementById("cart-items-number");
 const cartButton = document.getElementById("cart-button");
+const emptyCartModal = document.getElementById("empty-cart-modal");
+const cartModal = document.getElementById("cart-modal");
+const cartSummary = document.getElementById("cart-summary");
 
-let cartItems = 0;
+let cartItems = [];
+let selectedPack = {
+  name: "",
+  price: 0,
+  quantity: 0,
+  icon: "",
+};
+let cartClicked = false;
+
+const closeModal = () => {
+  confirmationModalOverlay.classList.add("opacity-0", "pointer-events-none");
+  confirmationModalOverlay.classList.remove("opacity-100");
+
+  darkOverlay.classList.replace("opacity-60", "opacity-0");
+};
+const toggleModalVisibility = (modal, isOpen) => {
+  if (isOpen) {
+    modal.classList.remove("scale-0", "opacity-0", "pointer-events-none");
+    modal.classList.add("scale-100", "opacity-100");
+  } else {
+    modal.classList.add("scale-0", "opacity-0", "pointer-events-none");
+    modal.classList.remove("scale-100", "opacity-100");
+  }
+};
+
+const calculateTotal = () => {
+  let total = 0;
+  cartItems.forEach((item) => {
+    total += item.price * item.quantity;
+  });
+  return total;
+};
+const updateTotalDisplay = () => {
+  const total = calculateTotal();
+  document.getElementById("total-cart-display").textContent = `$${total}`;
+};
 
 addToCartButtons.forEach((button) => {
   button.addEventListener("click", (event) => {
-    selectedPack = event.target.dataset.pack;
+    selectedPack.name = event.target.dataset.pack;
+    selectedPack.price = parseFloat(event.target.dataset.price);
+    selectedPack.icon = event.target.dataset.icon;
```

Comment:
try destructuring the dataset properties (pack, price, icon), making the code more concise and avoiding repeated lookups:
`const { pack: name, price, icon } = data.target.dataset;`
` Object.assign(selectedPack, { name, price: parseFloat(price), icon });`